microceph/ceph/maintenance.go

```diff
@@ -0,0 +1,104 @@
+package ceph
+
+import (
+	"fmt"
+)
+
+type Maintenance struct {
+	Node       string
+	ClusterOps ClusterOps
+}
+
+func (m *Maintenance) Exit(dryRun, checkOnly, ignoreCheck bool) ([]Result, error) {
+	// General sanity checks
+	if checkOnly && ignoreCheck {
+		err := fmt.Errorf("'check_only' and 'ignore_check' are mutually exclusive.")
+		return []Result{}, err
+	}
+
+	// Preflight checks for exiting maintenance mode (currently empty)
+	preflightChecks := []Operation{}
+
+	// Main operations
+	operations := []Operation{
+		&UnsetNooutOps{ClusterOps: m.ClusterOps},
+		&AssertNooutFlagUnsetOps{ClusterOps: m.ClusterOps},
+		&StartOsdOps{ClusterOps: m.ClusterOps},
+	}
+
+	// Execute the maintenance operations
+	results := []Result{}
+	if checkOnly {
+		// Only run preflight checks
+		results = append(results, RunOperations(m.Node, preflightChecks, dryRun, false)...)
+	} else if ignoreCheck {
+		// Only run main operations (ignore preflight checks)
+		results = append(results, RunOperations(m.Node, operations, dryRun, false)...)
+	} else {
+		// Run both preflight checks and main operations
+		results = append(results, RunOperations(m.Node, preflightChecks, dryRun, false)...)
+		// Return the result now if there's error in preflight checks
+		for _, result := range results {
+			if result.Error != "" {
+				return results, nil // the error is not for operation error
+			}
+		}
+		// Otherwise, continue with the main operations
+		results = append(results, RunOperations(m.Node, operations, dryRun, false)...)
+	}
+
+	return results, nil
+}
+
+func (m *Maintenance) Enter(force, dryRun, setNoout, stopOsds, checkOnly, ignoreCheck bool) ([]Result, error) {
+	// General sanity checks
+	if checkOnly && ignoreCheck {
+		err := fmt.Errorf("'check_only' and 'ignore_check' are mutually exclusive.")
```

Comment:
Spelling nit: should be `check-only` and `ignore-check`

Reply:
Fixed